From 9ac269fffc88682ea487c3d9efecc07ce3d72417 Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Tue, 3 Feb 2015 00:04:39 +0000
Subject: [PATCH 2/4] Orient before resize

An image is currently re-sized before being orientated. This results in the bottom of portrait-orientated pics going off-screen.
This patch orientates an image before re-sizing it.

diff --git a/mythtv/libs/libmythui/mythuiimage.cpp b/mythtv/libs/libmythui/mythuiimage.cpp
index 03dcc80..e6d321e 100644
--- a/mythtv/libs/libmythui/mythuiimage.cpp
+++ b/mythtv/libs/libmythui/mythuiimage.cpp
@@ -318,6 +318,17 @@ class ImageLoader
 
         if (image && !bFoundInCache)
         {
+            if (imProps.isReflected)
+                image->Reflect(imProps.reflectAxis, imProps.reflectShear,
+                               imProps.reflectScale, imProps.reflectLength,
+                               imProps.reflectSpacing);
+
+            if (imProps.isGreyscale)
+                image->ToGreyscale();
+
+            if (imProps.isOriented)
+                image->Orientation(imProps.orientation);
+
             // Even if an explicit size wasn't defined this image may still need
             // to be scaled because of a difference between the theme resolution
             // and the screen resolution. We want to avoid scaling twice.
@@ -359,17 +370,6 @@ class ImageLoader
                 image->setAlphaChannel(mask.alphaChannel());
             }
 
-            if (imProps.isReflected)
-                image->Reflect(imProps.reflectAxis, imProps.reflectShear,
-                               imProps.reflectScale, imProps.reflectLength,
-                               imProps.reflectSpacing);
-
-            if (imProps.isGreyscale)
-                image->ToGreyscale();
-
-            if (imProps.isOriented)
-                image->Orientation(imProps.orientation);
-
             if (!imageReader)
                 GetMythUI()->CacheImage(cacheKey, image);
         }
-- 
1.9.1

